eva-accession-clustering/src/main/java/uk/ac/ebi/eva/accession/clustering/batch/io/ClusteringWriter.java

```diff
@@ -165,185 +161,230 @@ private void getOrCreateClusteredVariantAccessions(List<? extends SubmittedVaria
+        Long variantAccession = clusteredVariantEntity.getAccession();
+        String variantHash = clusteredVariantEntity.getHashedMessage();
+        Long accessionInDB = allExistingHashesInDB.get(variantHash);
+
+        if (allExistingHashesInDB.containsKey(variantHash) && variantAccession != accessionInDB) {
+            if (mergeCandidateSVOE.containsKey(variantHash)) {
+                mergeCandidateSVOE.get(variantHash).getInactiveObjects()
+                        .add(new SubmittedVariantInactiveEntity(submittedVariantEntity));
+            } else {
+                SubmittedVariantOperationEntity submittedVariantOperationEntity = new SubmittedVariantOperationEntity();
+                List<SubmittedVariantInactiveEntity> inactiveObjects =
+                        Collections.singletonList(new SubmittedVariantInactiveEntity(submittedVariantEntity));
+                submittedVariantOperationEntity.fill(EventType.RS_MERGE_CANDIDATES, accessionInDB,
+                        "RS mismatch with " + accessionInDB, inactiveObjects);
 
-        if (!svToUpdate.isEmpty()) {
-            List<SubmittedVariantOperationEntity> operations =
-                    svToUpdate.stream()
-                              .map(sv -> buildSubmittedOperation(sv, newAccession))
-                              .collect(Collectors.toList());
-            mongoTemplate.insert(operations, submittedOperationCollection);
-            clusteringCounts.addSubmittedVariantsUpdateOperationWritten(operations.size());
+                mergeCandidateSVOE.put(variantHash, submittedVariantOperationEntity);
+            }
+            return true;
         }
+        return false;
     }
 
+    private boolean checkIfCandidateForRSSplit(SubmittedVariantEntity submittedVariantEntity,
+                                               ClusteredVariantEntity clusteredVariantEntity,
+                                               String assembly,
+                                               Map<Long, Set<String>> allExistingHashesGroupByRS,
+                                               Map<Long, SubmittedVariantOperationEntity> rsSplitCandidateSVOE) {
+        Long variantAccession = clusteredVariantEntity.getAccession();
+        String variantHash = clusteredVariantEntity.getHashedMessage();
+
+        if (!allExistingHashesGroupByRS.get(variantAccession).isEmpty() &&
+                !allExistingHashesGroupByRS.get(variantAccession).contains(variantHash)) {
+            if (rsSplitCandidateSVOE.containsKey(variantAccession)) {
+                List<SubmittedVariantInactiveEntity> inactiveEntities = rsSplitCandidateSVOE.get(variantAccession)
+                        .getInactiveObjects();
+                boolean submittedVariantAlreadyExist = inactiveEntities.stream()
+                        .map(inactiveObject -> inactiveObject.getModel())
+                        .anyMatch(sv -> sv.equals(submittedVariantEntity.getModel()));
```

Comment:
Why not just check the submitted variant accession?

Reply:
As discussed offline, leaving it as it is.